localsampletrack.go

```diff
@@ -209,6 +210,12 @@ func (s *LocalSampleTrack) WriteSample(sample media.Sample, opts *SampleWriteOpt
 		s.sequencer.NextSequenceNumber()
 	}
 
+	// calculate / interpolate duration when supplied duration is invalid
+	if sample.Duration.Nanoseconds() < 0 {
```

Comment:
should we be doing this when duration is set to 0 as well? Do you know what would cause duration to be negative?

Reply:
The stream causing this issue has `-1` duration set.  It might be OK if we check for `<= 0` but it isn't necessary for this specific issue.